app/src/main/java/com/kickstarter/ui/activities/compose/projectpage/AddOnsScreen.kt

```diff
@@ -0,0 +1,320 @@
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Card
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.material.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import com.kickstarter.R
+import com.kickstarter.libs.Environment
+import com.kickstarter.models.Project
+import com.kickstarter.models.Reward
+import com.kickstarter.models.ShippingRule
+import com.kickstarter.ui.compose.designsystem.KSPrimaryGreenButton
+import com.kickstarter.ui.compose.designsystem.KSTheme
+import com.kickstarter.ui.compose.designsystem.KSTheme.colors
+import com.kickstarter.ui.compose.designsystem.KSTheme.dimensions
+import com.kickstarter.ui.compose.designsystem.KSTheme.typography
+import com.kickstarter.ui.compose.designsystem.shapes
+
+@Composable
+@Preview(name = "Light", uiMode = Configuration.UI_MODE_NIGHT_NO)
+@Preview(name = "Dark", uiMode = Configuration.UI_MODE_NIGHT_YES)
+private fun AddOnsScreenPreview() {
+    KSTheme {
+        Scaffold(
+            backgroundColor = colors.backgroundAccentGraySubtle
+        ) { padding ->
+            AddOnsScreen(
+                modifier = Modifier.padding(padding),
+                environment = Environment.Builder().build(),
+                lazyColumnListState = rememberLazyListState(),
+                countryList = listOf(),
+                onShippingRuleSelected = {},
+                rewardItems = (0..10).map {
+                    Reward.builder()
+                        .title("Item Number $it")
+                        .description("This is a description for item $it")
+                        .id(it.toLong())
+                        .convertedMinimum((100 * (it + 1)).toDouble())
+                        .isAvailable(it != 0)
+                        .limit(if (it == 0) 1 else 10)
+                        .build()
+                },
+                project =
+                Project.builder()
+                    .currency("USD")
+                    .currentCurrency("USD")
+                    .build(),
+                onItemAddedOrRemoved = {},
+                onContinueClicked = {}
+            )
+        }
+    }
+}
+
+@Composable
+fun AddOnsScreen(
```

Comment:
Could we add tests? All parameters seem able to be mocked

Reply:
i was planning to add tests once we get to implementation because im not 100% sure ive covered all possible cases for these views yet.